src/components/date/__internal__/utils.test.ts

```diff
@@ -0,0 +1,448 @@
+  ["dd.MM.yyyy", "01.02.1900", "1900-02-01"],
+  ["dd,MM,yyyy", "01,02,1900", "1900-02-01"],
+  ["dd-MM-yyyy", "01-02-1900", "1900-02-01"],
+  ["dd/MM/yyyy", "01/02/1900", "1900-02-01"],
+  ["d.M.yyyy", "1.2.1900", "1900-02-01"],
+  ["d,M,yyyy", "1,2,1900", "1900-02-01"],
+  ["d-M-yyyy", "1-2-1900", "1900-02-01"],
+  ["d/M/yyyy", "1/2/1900", "1900-02-01"],
+  ["d,M,yy", "1,2,11", "2011-02-01"],
+  ["d-M-yy", "1-2-11", "2011-02-01"],
+  ["d/M/yy", "1/2/11", "2011-02-01"],
+  ["dd.M.yy", "01.2.11", "2011-02-01"],
+  ["dd,M,yy", "01,2,11", "2011-02-01"],
+  ["dd-M-yy", "01-2-11", "2011-02-01"],
+  ["dd/M/yy", "01/2/11", "2011-02-01"],
+  ["ddMMyy", "010223", "2023-02-01"],
+  ["dd.MM.yy", "01.02.23", "2023-02-01"],
+  ["dd-MM-yy", "01-02-23", "2023-02-01"],
+  ["dd/MM/yy", "01/02/23", "2023-02-01"],
+  ["d", "1", `${currentYear}-01-01`],
+  ["d.M", "1.2", `${currentYear}-02-01`],
+  ["d,M", "1,2", `${currentYear}-02-01`],
+  ["d-M", "1-2", `${currentYear}-02-01`],
+  ["d/M", "1/2", `${currentYear}-02-01`],
+  ["dd", "11", `${currentYear}-01-11`],
+  ["ddM", "112", `${currentYear}-02-11`],
+  ["dd.M", "11.2", `${currentYear}-02-11`],
+  ["dd,M", "11,2", `${currentYear}-02-11`],
+  ["dd-M", "11-2", `${currentYear}-02-11`],
+  ["dd/M", "11/2", `${currentYear}-02-11`],
+  ["d.MM", "1.04", `${currentYear}-04-01`],
+  ["d,MM", "1,04", `${currentYear}-04-01`],
+  ["d-MM", "1-04", `${currentYear}-04-01`],
+  ["d/MM", "1/04", `${currentYear}-04-01`],
+  ["ddMM", "0204", `${currentYear}-04-02`],
+  ["dd.MM", "02.04", `${currentYear}-04-02`],
+  ["dd,MM", "02,04", `${currentYear}-04-02`],
+  ["dd-MM", "02-04", `${currentYear}-04-02`],
+  ["dd/MM", "02/04", `${currentYear}-04-02`],
+])(
+  "should return the expected ISO formatted string when %s format and %s reference date is passed to `formatToISO`",
+  (formatString, referenceDate, expected) => {
+    expect(formatToISO(formatString, referenceDate)).toEqual(expected);
+  }
+);
+
+test("should return null if an invalid reference date is passed to `formatToISO`", () => {
+  expect(formatToISO("dd/MM/yyyy", "foo")).toEqual(null);
+});
+
+test.each(["ddMMyyyy", "dd.MM.yyyy", "dd,MM,yyyy", "dd-MM-yyyy", "dd/MM/yyyy"])(
+  "should return a valid formatted date string when a valid date and %s format is passed to `formattedValue`",
+  (format) => {
+    expect(formattedValue(format, new Date("2022-01-01"))).toEqual(
+      result(format)
+    );
+  }
+);
+
+test("should return an empty string when no date passed to `formattedValue`", () => {
+  expect(formattedValue("ddMMyyyy")).toEqual("");
+});
+
+test("should return the passed format and value if years in format string passed to `additionalYears` is not of length 2", () => {
+  expect(additionalYears("dd/MM/yyyy", "01/01/2022")).toEqual([
+    "dd/MM/yyyy",
+    "01/01/2022",
+  ]);
+
+  expect(additionalYears("yyyy/MM/dd", "2022/01/01")).toEqual([
+    "yyyy/MM/dd",
+    "2022/01/01",
+  ]);
+});
+
+describe.each(yearValuesLessThan69)(
```

Comment:
comment(non-blocking): this seems very wasteful and excessive

Looping through so many dates and making a new describe block for each is a little too much in my opinion. Maybe something we can chat about after this is in and make a ticket to address.

Reply:
Yeah the describe could be removed but I think we still need to loop or at least agree on a subset of dates to assert as there's specific logic in the component around how it treats abbreviated years. I think it's worth pushing the discussion and potential new ticket outside this PR as I suspect there's other components that involve user input that will have similar approaches to testing etc